### PR TITLE
Update ROCm recipes for rocm-5.1.0 release

### PR DESCRIPTION
**Pull request by srekolam**

### CODE REVIEW

File: var/spack/repos/builtin/packages/hipblas/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hipblas/package.py]:
```suggestion
                '5.0.2', '5.1.0']:
```

File: var/spack/repos/builtin/packages/rocm-cmake/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocm-cmake/package.py:16]: The version change isn't really needed here since Spack will automatically replace it.

File: var/spack/repos/builtin/packages/rocblas/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/rocblas/package.py:111]: Curious: Why use commits that don't (necessarily) line up with the commits for the associated releases?

File: var/spack/repos/builtin/packages/hip/package.py
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/hip/package.py:65]: This is the only checksum I couldn't confirm using `spack checksum`.

**Review by tldahlgren — CHANGES_REQUESTED** [review]
Confirmed all but one checksum.  Suggested a minor ordering change.

Will defer final approval @haampie.

File: var/spack/repos/builtin/packages/hipblas/package.py
Comment (srekolam) [reply var/spack/repos/builtin/packages/hipblas/package.py]: made the required changes.

**Comment by srekolam** [comment]
This PR will address majority of the rocm recipes (39/46). Few of recipes are pending for updates (miopen-hip, miopen-opencl,migraphx, mivisionx,rocm-tensile) . i will plan to update this PR or will submit a new PR soon.

**Comment by eugeneswalker** [comment]
We would love to have the 5.1.0 recipes in Spack by the time we release E4S 22.05. We plan to use the state of `spack@develop` as of mid-May. Do you think these ROCm updates will be in develop by then? Thank you!

**Comment by eugeneswalker** [comment]
I see all the tests here have passed -- @srekolam is there anything else that needs to be done before merging?

**Comment by srekolam** [comment]
@eugeneswalker , thanks. I have updated the branch to the latest spack:develop yesterday. I will followup with another PR that would have updates to remaining 5-6 rocm recipes ( miopen-hip, miopen-opencl, migraphx, mivisionx,rocm-tensile ) . Also, there is plan to release 5.1.2 in next couple of weeks. if that happens before your E4S release, then i can update these recipes with the latest 5.1.2

**Comment by srekolam** [comment]
added migraphx, miopen-hip, miopen-opencl, mivisionx, rocm-tensile.
the mivisionx backend=opencl (default) is building whereas mivisionx backend=HIP has build error which i am working to resolve. Will followup with a PR if possible on that.
Starting with 5.1.0 , miopen (hip and opencl ) variant depends on the MLIR dialect and i created a new recipe for that.

**Review by tldahlgren — APPROVED** [review]

**Comment by tldahlgren** [comment]
I will defer the decision to merge this PR to @haampie

File: var/spack/repos/builtin/packages/rocblas/package.py
Comment (cgmb) [reply var/spack/repos/builtin/packages/rocblas/package.py:111]: @srekolam is using the commit from [tensile_tag.txt](https://github.com/ROCmSoftwarePlatform/rocBLAS/blob/rocm-5.1.0/tensile_tag.txt), which specifies the version of Tensile that is downloaded and used as part of the normal rocBLAS build. For releases, tensile_tag.txt is supposed to be the same as the version of Tensile that is tagged, but it seems that our process is to update the change log as the very last step before tagging a release. Thus, tensile_tag.txt ends up slightly out-of-date.

It should be fine to use either the version tagged for release or the version from tensile_tag.txt. IMO, it's a minor process wart that they're different. They're intended to be identical. The changelog is the only difference.

**Comment by eugeneswalker** [comment]
@srekolam Is this ready for merge ?

**Comment by alalazo** [comment]
@spackbot maintainers

**Comment by srekolam** [comment]
@eugeneswalker  , yeah..i think so. I will be raising a separate PR for ROCm-5.1.2 when it is released.

**Comment by eugeneswalker** [comment]
@haampie Any objects to us merging this ?

**Review by tgamblin — APPROVED** [review]

**Pull request merged**
